install/installer/pkg/common/common.go

```diff
@@ -154,6 +154,41 @@ func tracingEnv(context *RenderContext, tracing *experimental.Tracing) (res []co
 		//			 but would make env var composition more cumbersome.
 	}
 
+	if context.Config.Observability.Tracing.SecretName != nil {
+		res = append(res, corev1.EnvVar{
+			Name: "JAEGER_USER",
+			ValueFrom: &corev1.EnvVarSource{SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: *context.Config.Observability.Tracing.SecretName},
+				Key:                  "JAEGER_USER",
+			}},
+		})
+
+		res = append(res, corev1.EnvVar{
+			Name: "JAEGER_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: *context.Config.Observability.Tracing.SecretName},
+				Key:                  "JAEGER_PASSWORD",
+			}},
+		})
+	}
```

Comment:
Why do we expose these as ENV variables? In general, mounting the secret as a volume is preferred for credentials and here we could do that if the SecretName is specified.

Reply:
That's how the [library is configured](https://github.com/jaegertracing/jaeger-client-go/blob/8d8e8fcfd04de42b8482476abac6a902fca47c18/README.md#environment-variables).

https://github.com/gitpod-io/gitpod/blob/45b0b45ec09ab90e09f24c620b56eef94ca52673/components/common-go/tracing/tracing.go#L42

Reply:
We could construct that manually, from the information we mount - https://pkg.go.dev/github.com/uber/jaeger-client-go/config#Configuration

I'll raise an issue as a follow-up (which will likely never get done) as I don't want to block your change but we do abuse env variables a fair bit.